src/SOS/Strike/gcroot.cpp

```diff
@@ -1947,31 +1946,34 @@ void replace(std::wstring &str, const WCHAR *toReplace, const WCHAR *replaceWith
         i = str.find(toReplace, i + replaceWithLen);
     }
 }
-#endif
 
-void HeapTraverser::PrintType(size_t ID,LPCWSTR name)
+void HeapTraverser::PrintType(size_t ID, LPCWSTR wname)
 {
     if (m_format==FORMAT_XML)
     {
-#ifndef FEATURE_PAL
+        int len = (int)_wcslen(wname);
+        int size = WideCharToMultiByte(CP_ACP, 0, wname, len, NULL, 0, NULL, NULL);
+        char *buffer = (char*)_alloca(size + 1);
+        WideCharToMultiByte(CP_ACP, 0, wname, len, buffer, size, NULL, NULL);
+        buffer[size] = '\0';
+
         // Sanitize name based on XML spec.
-        std::wstring wname = name;
-        replace(wname, W("&"), W("&amp;"));
-        replace(wname, W("\""), W("&quot;"));
-        replace(wname, W("'"), W("&apos;"));
-        replace(wname, W("<"), W("&lt;"));
-        replace(wname, W(">"), W("&gt;"));
-        name = wname.c_str();
-#endif
+        std::string name(buffer);
+        replace(name, "&", "&amp;");
+        replace(name, "\"", "&quot;");
+        replace(name, "'", "&apos;");
+        replace(name, "<", "&lt;");
+        replace(name, ">", "&gt;");
+
         fprintf(m_file,
-            "<type id=\"%d\" name=\"%S\"/>\n",
-            ID, name);
+            "<type id=\"%d\" name=\"%s\"/>\n",
+            ID, name.c_str());
```

Comment:
Have you tried creating a non-ascii type with a multibyte encoding char and see that it gets encoded correctly into the file?

Reply:
I haven't tested that. I didn't really think about multibyte encodings.